tests/test_integration.py

```diff
@@ -141,4 +141,51 @@ def test_cr_interface_func(self):
         with open(os.path.join(os.path.dirname(__file__), 'edf_algo_charging_rates.json'), 'r') as infile:
             self.edf_algo_true_cr = json.load(infile)
 
-        self.assertDictEqual(self.edf_algo_true_cr, self.sch.polled_charging_rates)
\ No newline at end of file
+        self.assertDictEqual(self.edf_algo_true_cr, self.sch.polled_charging_rates)
+
+class EmptyScheduler(BaseAlgorithm):
+    ''' Always submits an empty schedule (empty dict) as the output
+    of its run function.
+    '''
+    def __init__(self):
+        super().__init__()
+
+    def schedule(self, active_evs):
+        return {}
+
+class TestEmptyScheduleSim(TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.plugin_event = acnsim.PluginEvent(
+            10,
+            acnsim.EV(
+                10, 20, 30, 'PS-001', 'EV-001',
+                acnsim.Battery(100, 50, 20)
+            )
+        )
+
+        self.evse = acnsim.EVSE('PS-001', max_rate=32,
```

Comment:
Test with multiple EVSEs to find bug in line 207 where we use len(a) instead of a.shape[0].

Reply:
Done.